src/gui/entry/EntryModel.cpp

```diff
@@ -288,6 +310,26 @@ QVariant EntryModel::data(const QModelIndex& index, int role) const
                 return resources()->icon("chronometer");
             }
             break;
+        case PasswordStrength:
+            PasswordHealth health(entry->password());
```

Comment:
This gets recalculated on every paint event. We really need to cache this computation in Entry and invalidate the cache when a change is made to the entry. This would also be useful for the reports themselves as well.

Reply:
The health is now cached alongside any changes made to the password